lib/versioning.h

```diff
@@ -151,8 +151,8 @@
 
 /* config version code, in the same format as GlobalConfig->version */
 #define VERSION_VALUE   0x0319
-#define VERSION_CURRENT VERSION_3_25
```

Comment:
Is this intentionally removed?

Reply:
I agree with @alltilla, it's needed.
It is still used in code, that's the reason for the failed Travis jobs.
code:
```
lib/cfg.c-  if (cfg_is_config_version_older(self, VERSION_LAST_SEMANTIC_CHANGE))
lib/cfg.c-    {
lib/cfg.c-      msg_warning("WARNING: Configuration file format is too old, syslog-ng is running in compatibility mode. "
lib/cfg.c:                  "Please update it to use the " VERSION_CURRENT " format at your time of convenience. "
lib/cfg.c-                  "To upgrade the configuration, please review the warnings about incompatible changes printed "
lib/cfg.c-                  "by syslog-ng, and once completed change the @version header at the top of the configuration "
```

We should encourage to use the latest config version, even there were no semantic change between 2 versions.

Reply:
No. This warning should only appear if there is a real backwards incompatible change.

One of the reasons that distros don't allow upgrade syslog-ng is the change of version number in syslog-ng.conf and the warning coming with it. If syslog-ng warns only when there is a real change, we could do version upgrades instead of trying to hand pick patches.

Reply:
> This warning should only appear if there is a real backwards incompatible change.

I agree, and this is what this PR solves.
My saying is, that if the user uses a feature with an old config, say 3.18, and that feature changed with version 3.22, we should say that "although it changed with version 3.22, use version 3.25 instead".

Of course from config version point of view there is no difference between 3.22 and 3.25, but I think upgrading syslog-ng itself to 3.25 and then setting the config version to 3.22 gives a false sense of an LTS version.

Reply:
In `syslog-ng -V` output the "Config Version" is the latest release, so I guess this supports my argument.

https://github.com/syslog-ng/syslog-ng/blob/acdd456209f5a629a1e21f1bbdfd3f5cee4ae975/syslog-ng/main.c#L139-L143

```
$ sbin/syslog-ng -V
syslog-ng 3 (3.25.1.78.gd5fa5ef)
Config version: 3.25
Installer-Version: 3.25.1.78.gd5fa5ef
Revision: 
Compile-Date: Jan 17 2020 09:36:30
```